numpy/_core/defchararray.py

```diff
@@ -121,6 +121,13 @@ def equal(x1, x2):
     out : ndarray
         Output array of bools.
 
+    Examples
+    --------
+    >>> y = "aa "
```

Comment:
I think this needs a bit more discussion as to what the way forward is. I think that we should remove the `rstrip` logic from the `np.char` functions and let `chararray` handle it, so that `np.char.array(['aa ']) == np.char.array(['aa'])`, but also `np.char.equal(np.array(['aa ']), np.array(['aa'])) == [False]`. @seberg @ngoldbaum What do you think?

Reply:
I could remove this example you want me to, or should we add a note? @lysnikolaou

Reply:
We had a talk with @rgommers about it a few days ago and he suggested that we leave `np.char` as is for a few versions after the `np.strings` namespace is there and deprecate it afterwards. My question was more about the in-between. Let's actually maybe open an issue and discuss there? This can go as-is I think.

Reply:
I think we should leave these functions as-is but we're free to make the behavior of things that get added to `np.strings` make more sense. I agree the `rstrip` behavior here is at minimum surprising but it's not worth changing if we're eventually going to deprecate this code anyway. Of course we can't do that until there's an alternative that's been around for a few releases.